plugins/trace_api_plugin/test/test_extraction.cpp

```diff
@@ -354,4 +373,54 @@ BOOST_AUTO_TEST_SUITE(block_extraction)
       BOOST_REQUIRE_EQUAL(data_log.at(expected_entry.offset).get<block_trace_v0>(), expected_trace);
    }
 
+   BOOST_FIXTURE_TEST_CASE(onerror_transaction_block, extraction_test_fixture)
+   {
+      auto onerror_act = make_onerror_action( "alice"_n, 1 );
+      auto actt1 = make_action_trace( 0, onerror_act, "eosio.token"_n );
+      auto ptrx1 = make_packed_trx( { onerror_act } );
+
+      auto act2 = make_transfer_action( "bob"_n, "alice"_n, "0.0001 SYS"_t, "Memo!" );
+      auto actt2 = make_action_trace( 1, act2, "bob"_n );
+      auto transfer_trx = make_packed_trx( { act2 } );
+
+      auto onerror_trace = make_transaction_trace( ptrx1.id(), 1, 1, chain::transaction_receipt_header::executed,
+                              { actt1 } );
+      auto transfer_trace = make_transaction_trace( transfer_trx.id(), 1, 1, chain::transaction_receipt_header::soft_fail,
+                                                   { actt2 } );
+      onerror_trace->failed_dtrx_trace = transfer_trace;
+
+      signal_applied_transaction( onerror_trace, transfer_trx.get_signed_transaction() );
+
+      auto bsp1 = make_block_state( chain::block_id_type(), 1, 1, "bp.one"_n,
+            { chain::packed_transaction(transfer_trx) } );
+      signal_accepted_block( bsp1 );
+
+      const uint64_t expected_lib = 0;
+      const block_entry_v0 expected_entry { bsp1->id, 1, 0 };
+
+      const block_trace_v0 expected_trace { bsp1->id, 1, bsp1->prev(), chain::block_timestamp_type(1), "bp.one"_n,
+         {
+            {
+               ptrx1.id(),
```

Comment:
I'm missing something I think.  `ptrx.id()` is the id of a "virtual" packed transaction that has the `onerror` action right?  `transfer_trx.id()` would be the ID of the transaction that "failed".   So, I would have expected this to produce a trace with that ID and a status of `soft_failed`